js/ssai_pulse.js

```diff
@@ -1121,34 +1217,49 @@ OO.Ads.manager(function(_, $)
       }
       return urls;
     }, this);
-
+    
     /**
-     * Callback used when the duration of an ad has passed.
+     * Helper function to call impressions.
      * @private
-     * @method SsaiPulse#_adEndedCallback
+     * @method SsaiPulse#_handleImpressionCalls
+     * @param {object} curId3Object An object with the impressions data
      */
-    var _adEndedCallback = _.bind(function()
-    {
-      _clearAdDurationTimeout();
-      if (this.currentAd)
-      {
-        amc.notifyLinearAdEnded(this.currentAd.id);
-        amc.notifyPodEnded(this.currentAd.id);
-        _handleTrackingUrls(this.currentAd, ["firstQuartile", "midpoint", "thirdQuartile", "complete"]);
+    var _handleImpressionCalls = _.bind(function(curId3Object) {
+      if (!isId3ContainsStartedTime(curId3Object.time)) {
```

Comment:
For my own curiosity, does this mean that we are sending out the impression when we receive the first id3 tag after the playback starts? Does only the initial id3 tag to prompt ad playback contain the start time?